astropy/wcs/src/wcslib_wrap.c

```diff
@@ -257,7 +259,6 @@ PyWcsprm_init(
       return -1;
     }
 
-    note_change(self);
```

Comment:
Good catch, makes no sense to set flag to 0 if it's set to -1 in the following line 👍